Launch KeePassXC password generator popup from the extension

varjolintu opened the pull request:
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Explain large or complex code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" )
Add support for launching the password generator popup from the browser and returning the generated password.
Removes all password related settings from BrowserSettings class.

Fixes https://github.com/keepassxreboot/keepassxc/issues/6473.

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Manually.

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (change that adds functionality)

droidmonkey commented on src/browser/BrowserAction.cpp:
```diff
@@ -303,25 +304,19 @@ QJsonObject BrowserAction::handleGetLogins(const QJsonObject& json, const QStrin
-    arr.append(password);
-
-    const QString newNonce = incrementNonce(nonce);
+    connect(generator, &PasswordGeneratorWidget::appliedPassword, [=](QString password) {
```
Use the static QObject::connect to avoid making BrowserAction a QObject.

droidmonkey commented on src/browser/BrowserService.cpp:
```diff
@@ -313,6 +312,28 @@ QString BrowserService::getCurrentTotp(const QString& uuid)
+    }
+
+    raiseWindow();
+    return m_passwordGenerator.data();
```
This pointer passing is an antipattern, can we clean this flow up?

varjolintu commented:
Not working yet, even if I made the fixes. If the password dialog is closed, after that it doesn't fill anymore when launched again.

varjolintu commented:
Fixed a bug where the signal was connected multiple times (on every new request). Moved the lambda to the constructor so we can make sure it's only connected once.

droidmonkey commented:
This works REALLY well, great job @varjolintu. As a note, when you connect a lambda function to a signal, you need to provide a context (the third parameter in `connect`) to ensure the lambda is destroyed when the context is destroyed. I added it to BrowserService and BrowserAction calls for you.
